src/Interpreters/maskSensitiveInfoInQueryForLogging.cpp

```diff
@@ -0,0 +1,607 @@
+        {
+            if (!function.arguments)
+                return false;
+
+            const auto * expr_list = function.arguments->as<ASTExpressionList>();
+            if (!expr_list)
+                return false; /// return false because we don't want to validate query here
+
+            const auto & arguments = expr_list->children;
+            if (arg_idx >= arguments.size())
+                return false;
+
+            ASTPtr argument = arguments[arg_idx];
+            try
+            {
+                argument = evaluateConstantExpressionOrIdentifierAsLiteral(argument, context);
+            }
+            catch (...)
+            {
+                return false;
+            }
+
+            const auto & literal = assert_cast<const ASTLiteral &>(*argument);
+            if (literal.value.getType() != Field::Types::String)
+                return false;
+
+            *value = literal.value.safeGet<String>();
+            return true;
+        }
+
+        static bool tryGetEvaluatedConstDatabaseNameFromArgument(
+            const ASTFunction & function, size_t arg_idx, const ContextPtr & context, String * value)
+        {
+            if (!function.arguments)
+                return false;
+
+            const auto * expr_list = function.arguments->as<ASTExpressionList>();
+            if (!expr_list)
+                return false; /// return false because we don't want to validate query here
+
+            const auto & arguments = expr_list->children;
+            if (arg_idx >= arguments.size())
+                return false;
+
+            ASTPtr argument = arguments[arg_idx];
+            try
+            {
+                argument = evaluateConstantExpressionForDatabaseName(argument, context);
+            }
+            catch (...)
+            {
+                return false;
+            }
+
+            const auto & literal = assert_cast<const ASTLiteral &>(*argument);
+            if (literal.value.getType() != Field::Types::String)
+                return false;
+
+            *value = literal.value.safeGet<String>();
+            return true;
+        }
+
+        static void visitDictionaryDef(ASTDictionary & dictionary, Data & data)
+        {
+            if (!dictionary.source || !dictionary.source->elements)
+                return;
+
+            const auto * expr_list = dictionary.source->elements->as<ASTExpressionList>();
+            if (!expr_list)
+                return; /// return because we don't want to validate query here
+
+            const auto & elements = expr_list->children;
+
+            /// We replace password in the dictionary's definition:
+            /// SOURCE(CLICKHOUSE(host 'example01-01-1' port 9000 user 'default' password 'qwe123' db 'default' table 'ids')) ->
+            /// SOURCE(CLICKHOUSE(host 'example01-01-1' port 9000 user 'default' password '[HIDDEN]' db 'default' table 'ids'))
+            for (const auto & element : elements)
```

Comment:
```suggestion
            for (const auto & element : expr_list->children)
```

Reply:
I like `for (const auto & element : elements)` more ;)

Reply:
its a minor detail, we only use `elements` here, so I thought no need of having another variable. but I leave it up to you :)